soc/arm/st_stm32/stm32h7/soc_m7.c

```diff
@@ -80,7 +80,7 @@ static int stm32h7_init(const struct device *arg)
 	SystemCoreClock = 64000000;
 
 	/* Power Configuration */
-#ifdef SMPS
+#if defined(SMPS) && defined(CONFIG_POWER_SUPPLY_SMPS)
```

Comment:
Could be interesting, in case CONFIG_POWER_SUPPLY_SMPS is enabled and SMPS is not, to raise a compilation error.
(CONFIG_POWER_SUPPLY_SMPS  should not be set when SMPS is not supported)